libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -308,6 +311,51 @@ bool IRGeneratorForStatements::visit(FunctionCall const& _functionCall)
 			")\n";
 		break;
 	}
+    case FunctionType::Kind::Assert:
+    case FunctionType::Kind::Require:
+    {
+        solAssert(arguments.size() > 0, "Expected at least one parameter for require/assert");
+
+        for (auto& arg: arguments)
+            arg->accept(*this);
+
+        if (arguments.size() == 1)
+        {
+            m_code << Whiskers("if iszero(<cond>) { <function> }")
+                ("cond", m_context.variable(*arguments[0]))
+                (
+                    "function",
+                    FunctionType::Kind::Assert == functionType->kind() ?
+                        "invalid()" : "revert(0, 0)"
+                ).render();
+            break;
+        }
+
+        auto const errHash =
+            (u256(FixedHash<4>::Arith(FixedHash<4>(dev::keccak256("Error(string)")))) << (256 - 32));
+
+        ABIFunctions abiFunctions(m_context.evmVersion(), m_context.functionCollector());
+        string const encodeFunc = abiFunctions.tupleEncoder(
+            {arguments[1]->annotation().type},
+            {TypeProvider::array(DataLocation::Memory, true)}
+        );
+
+        m_code << Whiskers(R"(
+            if iszero(<cond>)
+            {
+                let fmp := mload(0x40)
```

Comment:
Shouldn't use `0x40` literally, but either call a function or retrieve it from the utils / context.